node-repository/src/main/java/com/yahoo/vespa/hosted/provision/persistence/CuratorCache.java

```diff
@@ -0,0 +1,14 @@
+package com.yahoo.vespa.hosted.provision.persistence;
+
+import com.yahoo.path.Path;
+
+import java.util.List;
+import java.util.Optional;
+
+public abstract class CuratorCache {
```

Comment:
I suggest calling this Session, making it an interface and moving it inside CuratorDatabase.

Then we could also rename "CuratorDatabaseCache" to just "Cache", to make the short name style consistent, and remove any imports of the nested names, so we read CuratorDatabase.Cache and CuratorDatabase.Session.

Reply:
Now I think it should be better.